go/types/set.go

```diff
@@ -217,7 +217,21 @@ func (s Set) IterAll(cb setIterAllCallback) {
 }
 
 func (s Set) Iterator() SetIterator {
-	return &setIterator{s: s, cursor: nil}
+	return s.IteratorAt(0)
+}
+
+func (s Set) IteratorAt(idx uint64) SetIterator {
+	return &setIterator{
+		cursor: newCursorAtIndex(s.seq, idx, false),
+		s:      s,
+	}
+}
+
+func (s Set) IteratorFrom(val Value) SetIterator {
+	return &setIterator{
+		cursor: newCursorAtValue(s.seq, val, false, false, false),
```

Comment:
We should enable readahead for the iterators. I missed it in the map iterator review too, could you fix those?